src/libsync/propagatedownload.cpp

```diff
@@ -451,19 +436,76 @@ void PropagateDownloadFile::startDownload()
     {
         SyncJournalDb::DownloadInfo pi;
         pi._etag = _item->_etag;
-        pi._tmpfile = tmpFileName;
+        pi._tmpfile = _tmpFileName;
         pi._valid = true;
         propagator()->_journal->setDownloadInfo(_item->_file, pi);
         propagator()->_journal->commit("download file start");
     }
 
+    if (isZsyncPropagationEnabled(propagator(), _item)) {
+        QFile file(propagator()->getFilePath(_item->_file));
+        if (file.open(QIODevice::ReadOnly)) {
+            QNetworkRequest req;
+            req.setPriority(QNetworkRequest::LowPriority);
+
+            QUrl zsyncUrl = Utility::concatUrlPath(propagator()->account()->url(),
+                                                   QLatin1String("/index.php/apps/files/api/v1/zsync"));
+            zsyncUrl = Utility::concatUrlPath(zsyncUrl, propagator()->_remoteFolder + _item->_file);
+
+            auto job = propagator()->account()->sendRequest("GET", zsyncUrl, req);
+            connect(job, &SimpleNetworkJob::finishedSignal, this, &PropagateDownloadFile::slotZsyncGetMetaFinished);
+            return;
+        } else {
+            qCInfo(lcZsyncGet)
+                << "No local copy of:"
+                << propagator()->getFilePath(_item->_file);
+            startDownloadNormal();
+            return;
+        }
+    }
+
+    startDownloadNormal();
+}
+
+
+void PropagateDownloadFile::slotZsyncGetMetaFinished(QNetworkReply *reply)
+{
+    int httpStatusCode = reply->attribute(QNetworkRequest::HttpStatusCodeAttribute).toInt();
+    if (httpStatusCode != 200) {
+        qCWarning(lcPropagateDownload)
```

Comment:
Is this really a warning or something that will regularly come up? I.e. does the server provide a zsync control file for each file?

Reply:
qCInfo is probably more appropriate, fixed.